File: Pack Track/Pack Track/Model/Runner.swift
//
//  Runner.swift
//  Pack Track
//
//  Created by Keegan Jebb on 2018-05-12.
//  Copyright © 2018 Keegan Jebb. All rights reserved.
//

import Foundation
import CoreLocation

class Runner {
    
    var runnerID: String = ""
    var name: String = ""
    var numRunsInGroup: Int = 0
    var connected: Bool = true
    
    //Optionals are given values when the person is tracking their run
    var location: CLLocationCoordinate2D?
    var distance: Double?
    var averagePace: Double?
    
    func copyRunnerData(for templateRunner: Runner) {
        
        self.runnerID = templateRunner.runnerID
        self.name = templateRunner.name
        self.numRunsInGroup = templateRunner.numRunsInGroup
        self.connected = templateRunner.connected
        self.location = templateRunner.location
        self.distance = templateRunner.distance
        self.averagePace = templateRunner.averagePace
    }
}

File: RunnersCircleAndroid/app/src/main/java/com/keeganjebb/runnerscircleandroid/model/Runner.kt
package com.keeganjebb.runnerscircleandroid.model

import com.keeganjebb.runnerscircleandroid.support.toDistanceString
import com.keeganjebb.runnerscircleandroid.support.toPaceString
import com.google.android.gms.maps.model.LatLng

class Runner {

    var runnerID: String = ""
    var name: String = ""
    var numRunsInGroup: Int = 0
    var connected: Boolean = true

    //Optionals are given values when the person is tracking their run
    var location: LatLng? = null
    var distance: Double? = null
    var averagePace: Double? = null

    fun copyRunnerData(templateRunner: Runner) {

        runnerID = templateRunner.runnerID
        name = templateRunner.name
        numRunsInGroup = templateRunner.numRunsInGroup
        connected = templateRunner.connected
        location = templateRunner.location
        distance = templateRunner.distance
        averagePace = templateRunner.averagePace
    }


    fun getMarkerSnippet(): String {

        var distanceString = ""
        if (distance != null) distanceString = distance!!.toDistanceString()

        var paceString = ""
        if (averagePace != null) paceString = averagePace!!.toPaceString()

        return distanceString + "\n" + paceString
    }



}